### ISSUE TITLE
`range_plus_one` also triggers if both sides have a plus one

### ISSUE THREAD

**Issue by Kriskras99** [rust-lang/rust-clippy#13138]
### Summary

When constructing ranges where both sides have a `+ 1` the lint suggestion makes the code less clear.

### Lint Name

range_plus_one

### Reproducer

I tried this code:

```rust
let mmap = b"Some memory contents";
let offset = 4;
let load = 6;
let bits_eight = &mmap[offset + 1..offset + load + 1];
```

I saw this happen:

```
warning: an inclusive range would be more readable
   --> simdvcd/src/lib.rs:469:49
    |
469 |                     let bits_eight = &self.mmap[self.offset + 1..self.offset + load + 1];
    |                                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: use: `(self.offset + 1)..=(self.offset + load)`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#range_plus_one
    = note: `-W clippy::range-plus-one` implied by `-W clippy::pedantic`
    = help: to override `-W clippy::pedantic` add `#[allow(clippy::range_plus_one)]`
```

I expected to see this happen:
No warning

### Version

```text
rustc 1.81.0-nightly (506985649 2024-07-20)
binary: rustc
commit-hash: 5069856495870486134dd2ca0b0e2516308c5c2a
commit-date: 2024-07-20
host: x86_64-unknown-linux-gnu
release: 1.81.0-nightly
LLVM version: 18.1.7
```
